repo/repo.go

```diff
@@ -91,17 +91,31 @@ func ListRepositories(workspace *rule.File) (repos []*rule.Rule, repoFileMap map
 		repoFileMap[repo.Name()] = workspace
 	}
 
-	for _, d := range workspace.Directives {
+	repos, repoIndexMap, repoFileMap, err = getRepositoriesInside(workspace, workspace.Path, repos, repoIndexMap, repoFileMap)
```

Comment:
Could we apply the logic within the recursion to the initial case as well? Something like:
```
func ListRepositories(workspace *rule.File) (repos []*rule.Rule, repoFileMap map[string]*rule.File, err error) {
	repoIndexMap := make(map[string]int)
	repoFileMap = make(map[string]*rule.File)
        if err := listRepositoriesHelper(repoIndexMap, repoFileMap, repos, workspacce); err != nil {
            return nil, nil, err
        }
        return repos, repoFileMap, nil
}
        
func listRepositoriesHelper(repoFile *rule.File, ...) error {
        // parse repos from file
	
        for _, d := range repoFile.Directives {
		switch d.Key {
		case "repository_macro":
			f, defName, err := parseRepositoryMacroDirective(d.Value)
			if err != nil {
				return nil, nil, err
			}
			f = filepath.Join(filepath.Dir(workspace.Path), filepath.Clean(f))
			macroFile, err := rule.LoadMacroFile(f, "", defName)
			if err != nil {
				return nil, nil, err
			} 
                         // recursive call here:
                         listRepositoriesHelper(f, ...)
      }
}
```